netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -97,22 +97,12 @@ public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
           }
 
           @Override
-          public void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
-            // Don't care.
-          }
-
-          @Override
-          @Deprecated
-          public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
```

Comment:
It was marked as deprecated when I did the sync to 4.1.CR.  I assume that was because its going away, so I left this TODO to ease updating.  The only relevant method in this handler seems to be handlerAdded(), so I replaced it with ChannelDuplexHandler per Eric's comment.